var/spack/repos/builtin/packages/py-pyprecice/package.py

```diff
@@ -50,16 +55,20 @@ class PyPyprecice(PythonPackage):
     depends_on("py-mpi4py", type=("build", "run"))
     depends_on("py-cython@0.29:", type=("build"))
 
-    phases = ['install_lib', 'build_ext', 'install']
+    phases = ['build_ext', 'install']
 
     def build_ext_args(self, spec, prefix):
         return [
             "--include-dirs=" + spec["precice"].headers.directories[0],
             "--library-dirs=" + spec["precice"].libs.directories[0]
         ]
 
-    def install(self, spec, prefix):
+    def install_args(self, spec, prefix):
+        args = super(PyPyprecice, self).install_args(spec, prefix)
```

Comment:
This line calls the `install_args` method from the super class and stores the result in `args`. But the following line overwrites the value of `args`, discarding this. From looking at the setup.py, it seems like it does use `setuptools`, so I'm surprised that the default flags don't work. The flags we inject are responsible for the library being installed to the correct location, so they are kind of important.

Reply:
Yes, it is not perfect, but to my understanding that is the same as in the previous Spack package where I directly manipulated the `install` routine. I found it to be more descriptive actually work on the arguments. I also was not sure if I would mess up more than necessary when manipulating the `install` step directly. 

This was necessary since the installation routine `2.0.0.1:2.1.1.99` is missing some of the "standard" installation options that Spack wants to use. Especially `--single-version-externally-managed` is missing. See here an example output for `2.0.0.2`:
```
> python3 setup.py install --help
Common commands: (see '--help-commands' for more)

  setup.py build      will build the package underneath 'build/'
  setup.py install    will install the package

Global options:
  --verbose (-v)  run verbosely (default)
  --quiet (-q)    run quietly (turns verbosity off)
  --dry-run (-n)  don't actually do anything
  --help (-h)     show detailed help message
  --no-user-cfg   ignore pydistutils.cfg in your home directory

Options for 'my_install' command:
  --prefix            installation prefix
  --exec-prefix       (Unix only) prefix for platform-specific files
  --home              (Unix only) home directory to install under
  --install-base      base installation directory (instead of --prefix or --
                      home)
  --install-platbase  base installation directory for platform-specific files
                      (instead of --exec-prefix or --home)
  --root              install everything relative to this alternate root
                      directory
  --install-purelib   installation directory for pure Python module
                      distributions
  --install-platlib   installation directory for non-pure module distributions
  --install-lib       installation directory for all module distributions
                      (overrides --install-purelib and --install-platlib)
  --install-headers   installation directory for C/C++ headers
  --install-scripts   installation directory for Python scripts
  --install-data      installation directory for data files
  --compile (-c)      compile .py to .pyc [default]
  --no-compile        don't compile .py files
  --optimize (-O)     also compile with optimization: -O1 for "python -O", -O2
                      for "python -OO", and -O0 to disable [default: -O0]
  --force (-f)        force installation (overwrite any existing files)
  --skip-build        skip rebuilding everything (for testing/debugging)
  --record            filename in which to record list of installed files
  --install-layout    installation layout to choose (known values: deb, unix)
  --user              install in user site-package
                      '/home/jaustar/.local/lib/python3.6/site-packages'

usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help
```
If you would like to dig into this yourself that would be really great! :smile:  If you do not find the time, I would also happy with any feedback or hint what I would have to look into to make it work. I am not so experienced with `setuptools` and the Python toolchains (yet). I am a bit lost on where to find the correct information.